File: binding/src/libqflags.cpp
#include <QFlag>
#include <QIncompatibleFlag>
#include <qflags.h>
#include "libqflags.h"
#include "libqflags.hxx"

QFlag* QFlag_new(const QFlag* other) {
	 return new QFlag(*other);
}

QFlag* QFlag_new2(QFlag* other) {
	 return new QFlag(*other);
}

QFlag* QFlag_new3(int value) {
	 return new QFlag(value);
}

QFlag* QFlag_new4(const QFlag* param1) {
	 return new QFlag(*param1);
}

void QFlag_CopyAssign(QFlag* self, QFlag* other) {
    *self = *other;
}

void QFlag_MoveAssign(QFlag* self, QFlag* other) {
    *self = std::move(*other);
}

int QFlag_Operatorint(const QFlag* self) {
	return self->operator int();
}

void QFlag_Delete(QFlag* self) {
    delete self;
}

QIncompatibleFlag* QIncompatibleFlag_new(const QIncompatibleFlag* other) {
	 return new QIncompatibleFlag(*other);
}

QIncompatibleFlag* QIncompatibleFlag_new2(QIncompatibleFlag* other) {
	 return new QIncompatibleFlag(*other);
}

QIncompatibleFlag* QIncompatibleFlag_new3(int i) {
	 return new QIncompatibleFlag(i);
}

QIncompatibleFlag* QIncompatibleFlag_new4(const QIncompatibleFlag* param1) {
	 return new QIncompatibleFlag(*param1);
}

void QIncompatibleFlag_CopyAssign(QIncompatibleFlag* self, QIncompatibleFlag* other) {
    *self = *other;
}

void QIncompatibleFlag_MoveAssign(QIncompatibleFlag* self, QIncompatibleFlag* other) {
    *self = std::move(*other);
}

int QIncompatibleFlag_Operatorint(const QIncompatibleFlag* self) {
	return self->operator int();
}

void QIncompatibleFlag_Delete(QIncompatibleFlag* self) {
    delete self;
}

File: binding/src/libqflags.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qflag.html
pub const qflag = struct {

    /// New constructs a new QFlag object.
    pub fn New(other: ?*anyopaque) QtC.QFlag {
        return qtc.QFlag_new(@ptrCast(other));
    }


    /// New2 constructs a new QFlag object.
    pub fn New2(other: ?*anyopaque) QtC.QFlag {
        return qtc.QFlag_new2(@ptrCast(other));
    }


    /// New3 constructs a new QFlag object.
    pub fn New3(value: i32) QtC.QFlag {
        return qtc.QFlag_new3(value);
    }


    /// New4 constructs a new QFlag object.
    pub fn New4(param1: ?*anyopaque) QtC.QFlag {
        return qtc.QFlag_new4(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QFlag_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QFlag_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Operatorint(self: ?*anyopaque, ) i32 {
        return qtc.QFlag_Operatorint(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QFlag_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qincompatibleflag.html
pub const qincompatibleflag = struct {

    /// New constructs a new QIncompatibleFlag object.
    pub fn New(other: ?*anyopaque) QtC.QIncompatibleFlag {
        return qtc.QIncompatibleFlag_new(@ptrCast(other));
    }


    /// New2 constructs a new QIncompatibleFlag object.
    pub fn New2(other: ?*anyopaque) QtC.QIncompatibleFlag {
        return qtc.QIncompatibleFlag_new2(@ptrCast(other));
    }


    /// New3 constructs a new QIncompatibleFlag object.
    pub fn New3(i: i32) QtC.QIncompatibleFlag {
        return qtc.QIncompatibleFlag_new3(i);
    }


    /// New4 constructs a new QIncompatibleFlag object.
    pub fn New4(param1: ?*anyopaque) QtC.QIncompatibleFlag {
        return qtc.QIncompatibleFlag_new4(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QIncompatibleFlag_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QIncompatibleFlag_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Operatorint(self: ?*anyopaque, ) i32 {
        return qtc.QIncompatibleFlag_Operatorint(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QIncompatibleFlag_Delete(@ptrCast(self));
    }
};
